runtime/hydradx/src/xcm.rs

```diff
@@ -399,17 +440,34 @@ use xcm_executor::traits::ConvertLocation;
 /// Converts Account20 (ethereum) addresses to AccountId32 (substrate) addresses.
 pub struct EvmAddressConversion<Network>(PhantomData<Network>);
 impl<Network: Get<Option<NetworkId>>> ConvertLocation<AccountId> for EvmAddressConversion<Network> {
-	fn convert_location(location: &MultiLocation) -> Option<AccountId> {
+	fn convert_location(location: &Location) -> Option<AccountId> {
+		let Location { parents, interior } = location;
+		match interior {
+			Junctions::X1(a) if *parents == 0 => {
+				let j = a.as_ref()[0];
+				match j {
+					AccountKey20 { network: _, key } => {
+						let account_32 = ExtendedAddressMapping::into_account_id(H160::from(key));
+						Some(account_32)
+					}
+					_ => None,
+				}
+			}
+			_ => None,
+		}
+		// Note: keeping the original code for reference until tests are successful
```

Comment:
delete

Reply:
marking it back to unresolved as it was not deleted. Of we still have failing tests so we need it?!

Reply:
Oh, Thanks. Deleted.